template/SpellAction.sol

```diff
@@ -1,55 +1,82 @@
+import "lib/dss-interfaces/src/dss/FaucetAbstract.sol";
+import "lib/dss-interfaces/src/dss/GemJoinAbstract.sol";
+import "lib/dss-interfaces/src/dss/OsmMomAbstract.sol";
+import "lib/dss-interfaces/src/dss/OsmMomAbstract.sol";
+import "lib/dss-interfaces/src/dss/MedianAbstract.sol";
+import "lib/dss-interfaces/src/dss/DssAutoLineAbstract.sol";
+
 contract SpellAction {
-    address constant MCD_VAT = ;
-    address constant MCD_CAT = ;
-    address constant MCD_JUG = ;
-    address constant MCD_SPOT = ;
-    address constant MCD_POT = ;
-    address constant MCD_END = ;
-    address constant FLIPPER_MOM = ;
-    address constant OSM_MOM = ; // Only if PIP_TOKEN = Osm
-    address constant ILK_REGISTRY = ;
-    address constant FAUCET = ;
+    // KOVAN ADDRESSES
+    //
+    // The contracts in this list should correspond to MCD core contracts, verify
+    //  against the current release list at:
+    //     https://changelog.makerdao.com/releases/kovan/active/contracts.json
+    ChainlogAbstract constant CHANGELOG =
+        ChainlogAbstract(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
 
     address constant TOKEN = ;
     address constant MCD_JOIN_TOKEN_LETTER = ;
     address constant MCD_FLIP_TOKEN_LETTER = ;
     address constant PIP_TOKEN = ;
+    bytes32 constant ILK_TOKEN_A = "TOKEN-LETTER";
 
+    // decimals & precision
     uint256 constant THOUSAND = 10**3;
-    uint256 constant MILLION = 10**6;
-    uint256 constant WAD = 10**18;
-    uint256 constant RAY = 10**27;
-    uint256 constant RAD = 10**45;
+    uint256 constant MILLION  = 10**6;
+    uint256 constant WAD      = 10**18;
+    uint256 constant RAY      = 10**27;
+    uint256 constant RAD      = 10**45;
 
     // Many of the settings that change weekly rely on the rate accumulator
     // described at https://docs.makerdao.com/smart-contract-modules/rates-module
     // To check this yourself, use the following rate calculation (example 8%):
     //
     // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
     //
+    // A table of rates can be found at
+    //    https://ipfs.io/ipfs/QmefQMseb3AiTapiAKKexdKHig8wroKuZbmLtPLv4u2YwW
+    //
     uint256 constant public X_PERCENT_RATE = ;
 
     function execute() external {
-        bytes32 ilk = "TOKEN-LETTER";
+        address MCD_VAT      = CHANGELOG.getAddress("MCD_VAT");
+        address MCD_CAT      = CHANGELOG.getAddress("MCD_CAT");
+        address MCD_JUG      = CHANGELOG.getAddress("MCD_JUG");
+        address MCD_SPOT     = CHANGELOG.getAddress("MCD_SPOT");
+        address MCD_POT      = CHANGELOG.getAddress("MCD_POT");
+        address MCD_END      = CHANGELOG.getAddress("MCD_END");
+        address FLIPPER_MOM  = CHANGELOG.getAddress("FLIPPER_MOM");
+        address OSM_MOM      = CHANGELOG.getAddress("OSM_MOM"); // Only if PIP_TOKEN = Osm
+        address ILK_REGISTRY = CHANGELOG.getAddress("ILK_REGISTRY");
+        address FAUCET       = CHANGELOG.getAddress("FAUCET");
```

Comment:
Might be okay for kovan but for mainnet we should use the address file pattern that @brianmcmichael started.